### PR TITLE
Add org.alpine.linux

### PR DESCRIPTION
**Pull request by paledega**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by paledega** [comment]
This application provide virtual terminal from alpine linux with fakeroot permission provided by proot. (like termux)

It is unofficial project.

**Comment by paledega** [comment]
image
image

**Comment by paledega** [comment]
bot, build org,alpine.linux

File: launcher.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Version=1.0
+Name=Alpine
+GenericName=Alpine Linux
+Comment=Alpine subsystem
+Exec=/app/bin/alpine
```
Comment (hfiguiere) [inline launcher.desktop:7]:
```suggestion
Exec=alpine
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.alpine.linux.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+app-id: org.alpine.linux
```
Comment (hfiguiere) [inline org.alpine.linux.yaml:1]: also I'm uncertain about this app-id fitting the requirements.

**Comment by paledega** [comment]
Thank you. Merged :D

File: org.alpine.linux.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+sdk: org.freedesktop.Sdk
+command: /app/bin/alpine
+build-options:
+  build-args:
+    - --share=network
+finish-args:
+  - --talk-name=org.freedesktop.Flatpak
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=host
```
[the remark this answers is not in the document]
Comment (paledega) [reply org.alpine.linux.yaml:13]: --talk-name for running bubblewrap from host
--socket x11 and wayland for gui
--filesystem is binding /host directory

File: org.alpine.linux.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  - --share=network
+  - --device=dri
+  - --socket=pulseaudio
+  - --allow=bluetooth
```
[the remark this answers is not in the document]
Comment (paledega) [reply org.alpine.linux.yaml:17]: network is internet access
dri for gpu 
pulseaudio audio support

File: org.alpine.linux.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.alpine.linux.yaml:1]: In a second tought, it does NOT meet the requirement. `alpine.org` is definitely unrelated.

File: org.alpine.linux.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+app-id: org.alpine.linux
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: /app/bin/alpine
```
Comment (hfiguiere) [inline org.alpine.linux.yaml:5]:
```suggestion
command: alpine
```

File: org.alpine.linux.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+build-options:
+  build-args:
+    - --share=network
```
[the remark this answers is not in the document]
Comment (paledega) [reply org.alpine.linux.yaml:8]: I need fakeroot or root permission for generating alpine rootfs. flatpak builder is not supported root build. so I am using tarball from my github releases.

File: org.alpine.linux.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  - --filesystem=host
+  - --share=network
+  - --device=dri
+  - --socket=pulseaudio
+  - --allow=bluetooth
+modules:
+  - name: build
+    buildsystem: simple
+    build-commands:
+      - wget https://github.com/paledega/alpine-rootfs/releases/download/latest/alpine.tar.gz -O /app/alpine.tar.gz
```
[the remark this answers is not in the document]
Comment (paledega) [reply org.alpine.linux.yaml:21]: Hmm. I will publish as fixed release.

File: org.alpine.linux.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+      - install -D alpine.sh /app/bin/alpine
+      - install -D launcher.desktop  /app/share/applications/org.alpine.linux.desktop
+      - install -D icon.svg /app/share/icons/hicolor/scalable/apps/org.alpine.linux.svg
+      - bash install-proot.sh
```
[the remark this answers is not in the document]
Comment (paledega) [reply org.alpine.linux.yaml:25]: I will move all resources and scripts into git repository.

File: org.alpine.linux.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.alpine.linux.yaml:13]: I don't see anything even using x11 nor wayland.

File: org.alpine.linux.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.alpine.linux.yaml:21]: wget WILL NOT WORK. I reiterate that. There is NO NETWORK.

File: org.alpine.linux.yaml (hunk 2 above)
Comment (paledega) [reply org.alpine.linux.yaml:1]: I will rename to com.paledega.alpinelinux  because It is not official application.

**Comment by hfiguiere** [comment]
The app-id doesn't meet the requirements.

File: org.alpine.linux.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.alpine.linux.yaml:8]: flatpak-builder doesn't allow build-args on the builder. So it will not work as it is not allowed.

**Comment by paledega** [comment]
I will rename to com.paledega.alpinelinux because It is not official application.

**Comment by paledega** [comment]
@hfiguiere I recreate manifest.

**Comment by hfiguiere** [comment]
you didn't need to do that. You just wiped all previous review comments making reviewer life harder.

**Pull request closed without merging**